### DISCUSSION TITLE
Autoadd feature?

### DISCUSSION THREAD

**Discussion by nopoz** [trossr32/sonarr-radarr-lidarr-autosearch-browser-extension#198] [Ideas]
First off, thanks for this extension! 

Would be great if there could be a one-click-add feature. Clicking the arrs button on a supported page would auto-add via the API without any further interaction needed. In the configuration, the user would have to setup the defaults passed to the API for each app:

- Root Folder
- Monitor
- Minimum Availability
- Quality Profile
- Tags (Optional)
- Start search

**Comment by keldian** [comment]
Exactly what I was thinking after trying out this extension today. Frankly I don't see the point of sending you to an arr search, instead of just adding the title to the library directly via a POST payload. Why make me search when I already found the title? The most popular websites like IMDb and Trakt already have the ids baked in to the pages, which the arrs can parse to get the correct title.

**Comment by trossr32** [comment]
I think it would be more annoying to have to delete the inevitable incorrect additions than it is confirming which entity you want from the search page of your particular `arr`. Sure some integrations are more concrete, such as imdb or tvdb with their associated ids, but most of the integrations are based around looser text searches against the `arr` service. 

I guess there could be an option to be able to do a direct add via API _if_ an id has been used as the search query, but I'm not convinced it's worth the effort.

That all being said, I'm happy for features of this extension to be community developed and maintained, so if you'd like to add this feature behind a config flag in the settings then go for it and I'll happily deploy for everyone.

**Reply by imthenachoman** [reply]
If you are on an IMDB page for something, why/how would it send the wrong thing? It uses the IMDB ID, which is unique, right?
